src/index.ts

```diff
@@ -90,63 +98,34 @@ export function startServerAndCreateLambdaHandler<TContext extends BaseContext>(
   };
 }
 
-function normalizeGatewayEvent(event: GatewayEvent): HTTPGraphQLRequest {
-  if (isV1Event(event)) {
-    return normalizeV1Event(event);
-  }
-
-  if (isV2Event(event)) {
-    return normalizeV2Event(event);
+function normalizeGatewayEvent(event: IncomingEvent): HTTPGraphQLRequest {
+  let httpMethod: string;
+  if ('httpMethod' in event) {
+    httpMethod = event.httpMethod;
+  } else {
+    httpMethod = event.requestContext.http.method;
   }
-
-  throw Error('Unknown event type');
-}
-
-function isV1Event(event: GatewayEvent): event is APIGatewayProxyEvent {
-  // APIGatewayProxyEvent incorrectly omits `version` even though API Gateway v1
-  // events may include `version: "1.0"`
-  return (
-    !('version' in event) || ('version' in event && event.version === '1.0')
-  );
-}
-
-function isV2Event(event: GatewayEvent): event is APIGatewayProxyEventV2 {
-  return 'version' in event && event.version === '2.0';
-}
-
-function normalizeV1Event(event: APIGatewayProxyEvent): HTTPGraphQLRequest {
   const headers = normalizeHeaders(event.headers);
-  const body = parseBody(event.body, headers.get('content-type'));
-  // Single value parameters can be directly added
-  const searchParams = new URLSearchParams(
-    normalizeQueryStringParams(event.queryStringParameters),
-  );
-  // Passing a key with an array entry to the constructor yields
-  // one value in the querystring with %2C as the array was flattened to a string
-  // Multi values must be appended individually to get the to-spec output
-  for (const [key, values] of Object.entries(
-    event.multiValueQueryStringParameters ?? {},
-  )) {
-    for (const value of values ?? []) {
-      searchParams.append(key, value);
-    }
+  let search: string;
+  if ('rawQueryString' in event) {
+    search = event.rawQueryString;
+  } else if ('queryStringParameters' in event) {
+    event.queryStringParameters;
```

Comment:
stray line (looks like you have a similar workflow as me when checking types 😉)

Reply:
Haha! Seems like I do 😄